pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -57,25 +58,33 @@ type message struct {
 // provider contains dependencies for the DID exchange protocol and is typically created by using aries.Context()
 type provider interface {
 	OutboundDispatcher() dispatcher.Outbound
+	DIDWallet()          wallet.DIDCreator
 }
 
 // Service for DID exchange protocol
 type Service struct {
+	ctx             context
+	store           storage.Store
+	callbackChannel chan didCommChMessage
+	actionEvent     chan<- event.DIDCommEvent
+	statusEvents    []chan<- dispatcher.DIDCommMsg
+	lock            sync.RWMutex
+	statusEventLock sync.RWMutex
+	execute         bool
+}
+
+type context struct {
 	outboundDispatcher dispatcher.Outbound
-	store              storage.Store
-	callbackChannel    chan didCommChMessage
-	actionEvent        chan<- event.DIDCommEvent
-	statusEvents       []chan<- dispatcher.DIDCommMsg
-	lock               sync.RWMutex
-	statusEventLock    sync.RWMutex
-	execute            bool
+	didWallet          wallet.DIDCreator
 }
 
 // New return didexchange service
 func New(store storage.Store, prov provider) *Service {
 	svc := &Service{
-		outboundDispatcher: prov.OutboundDispatcher(),
-		store:              store,
+		ctx: context{
```

Comment:
why we can not just keep  `provider` instead of `ctx` ?

Reply:
@soluchok internal context was added based on this comment https://github.com/hyperledger/aries-framework-go/pull/267#discussion_r325940560

Reply:
ok